### PR TITLE
Create user page

### PR DESCRIPTION
**Pull request by zenithyap**
1. Create user profile page.
2. API features to get and edit user data.

### CODE REVIEW

File: front-end/peer-prep/src/index.tsx [hunk 1]
```diff
@@ -61,6 +62,7 @@ root.render(
                     </Route>
                     {/* All non-protected routes are written here */}
                     <Route path="SignIn" element={<SignInPage/>} />
+                    <Route path="/profile" element={<UserPage/>}/>
```
Comment (Acerizm) [inline front-end/peer-prep/src/index.tsx]: Please move this route under the `protected routes` under line 59 as only authenticated users should have access to the `UserPage`!

**Review by Acerizm — CHANGES_REQUESTED** [review]
LGTM! Thank you for creating the user page! A small change needs to be made for changing the routing of the `UserPage` and all will be good!

File: front-end/peer-prep/src/index.tsx (hunk 1 above)
Comment (zenithyap) [reply front-end/peer-prep/src/index.tsx]: Alright done, thanks for the feedback! Also, I changed the path to /user to keep the naming consistent.

Comment (Acerizm) [reply front-end/peer-prep/src/index.tsx]: Thank you too for keeping the naming consistent!

**Review by Acerizm — APPROVED** [review]

**Pull request merged**